Cursed heart

RemieRichards opened the pull request:
## Normal version:
- Adds a "Cursed Heart", a heart where you have to manually pump blood or you die
- Must be pumped every 3 seconds (you can pump 1 second early, to account for lag)
- A successful pump returns half the blood lost from a failed pump (so you can make up for failures)
- The screen turns red when you miss a pump, and remains red until you pump again
- Takes the singlebeat.ogg heart sound effect from Bay for use here, thanks!
- Exists as a punishment/gimmick item
## Wizard version:
- Wizards can buy a special cursed heart for 1 point (1 point as its dangerous)
- The wizard cursed heart heals 25 Brute/Burn/Oxy per pump
- The wizard cursed heart needs pumping every 6 seconds
## Misc:
- Lich now dumps all organs when moving to a new body (so the heart can be obtained again, by the wizard OR a crewmember)
- Fixes @phil235 trying to grant organ actions... to the organ. Lol, a 1 letter typo in a huge refactor isn't bad.

<del>Will conflict with @phil235's action button cleanup but I'm happy to fix conflicts.</del> Refactor Merged!

:cl: 
add: Cursed heart item for wizards, allows them to heal themselves in exchange for having to MANUALLY beat their heart every 6 seconds. Heals 25 brute/burn/oxy damage per correctly timed pump.
/:cl:

RemieRichards committed:
Adds a "Cursed Heart", a heart where you have to manually pump blood or you die.

ChangelingRain commented:
Fun fact; if you get a demon heart, you can use it to replace this cursed heart.

RemieRichards commented:
@ChangelingRain Yep! that or steal someone else's heart and give them this one.

PKPenguin321 commented on code/modules/surgery/organs/organ_internal.dm:
```diff
@@ -113,6 +133,39 @@
+/datum/action/item_action/organ_action/cursed_heart
+	name = "pump your blood"
+
+//You are now brea- pumping blood manually
```
REEEEEEEEEEEEEEE

PKPenguin321 commented:
so this is admin only, then?
i could see it being a wizard spell/event or maybe an away mission curse or something

RemieRichards commented:
@PKPenguin321 I was considering Wizards, but it seems too cruel.
I was also considering a subtype for wizards, where it gives them the heart, BUT it has more leniancy on the gaps between pumps, and each pump heals you so the wizard has to decide if they can keep it up in exchange for a decent heal, if that sounds cool, I'll add it.

ChangelingRain commented:
That sounds good. Maybe a visual indicator when it's getting close to needing another pump?

Buggy123 commented:
Won't this be a problem if the user gets stunned/incapacitated for any reason?

RemieRichards commented:
@Buggy123 yes, its why the default one is a punishment/joke item.

Buggy123 commented:
I know, just pointing out that beyond the annoying of constantly pumping your heart this effectively turns stuns into unavoidable death.

RemieRichards commented:
@Buggy123 the Wizard variant has 6 seconds (vs 3) so it should be a little fairer, plus I'm making it so successful pumps grant you (some) blood back, so if you fuck up you can still come back from it.

Buggy123 commented:
This is kind of weird though, wouldn't it effectively be easier to remove the heart completely and take salbutamol, etc than to keep the heart?

Without a heart, you just have oxyloss iirc, with the heart you lose blood constantly which will m.death() you if you lose too much.

RemieRichards commented:
@Buggy123 The Wizard version heals 25 Brute/Burn/Oxy damage per pump, if you can keep pumping it up thats a pretty decent heal, the idea is for it to be double-edged.

as for the normal heart, yeah you should get it removed.

RemieRichards committed:
Added a wizard version of the heart, it heals the user every pump and only requires pumping every 6 seconds (vs 3)
Pumping successfully now returns half of the blood lost from a missing pump (anymore and I doubt you'd die)
You can no longer spam the pumping (healing made this a requirement)

Xhuis commented:
I feel like this should pump automatically if the client disconnects or the mob is catatonic/brain-dead. Dying because your connection dropped doesn't seem very fun.

Jalleo commented:
What Xhuis said although remove healing for wizard heart if that happens.

Supermichael777 commented:
when i saw the title I actually though we were going to get a howels moving castle reference. Awww

RemieRichards commented:
I can add anti-logout stuff sure.
@Supermichael777 As much as I love that movie that'd be much harder to squeeze in. Atm this is a ~rough~ reference to Crypt of the Necrodancer, but with a spin that makes it a unique item for wizards/punishments.

RemieRichards committed:
Adds disconnect protection, I guess I'm not *that* mean.

RemieRichards commented on code/modules/spells/spell_types/lichdom.dm:
```diff
@@ -87,6 +87,9 @@
 					var/mob/living/carbon/C = old_body
 					for(var/obj/item/W in C)
 						C.unEquip(W)
+					for(var/obj/item/organ/internal/I in C.internal_organs)
```
Before I get any comments on this, things placed into people during operations (Nuke disks, clown masks, bombs, and so on) go into this list. They probably shouldn't, but they do, so I can't do my happy istypeless for loops here ;_;

RemieRichards committed:
Merge conflict fixes, Uses newly refactor actions, All cursed hearts can now be inserted by hand, not just wizard ones. Fixes organ actions trying to be granted to the object, not the user.

Merge branch 'master' of https://github.com/tgstation/-tg-station into cursed_heart

Conflicts:
	code/modules/surgery/organs/organ_internal.dm

RemieRichards commented:
@tgstation/commit-access this is ready now.
